acquia_lift.admin.wizard.inc

```diff
@@ -643,3 +807,191 @@ function _acquia_lift_personalize_campaign_wizard_variations_element(&$form, &$f
     );
   }
 }
+
+/**
+ * Helper function to generate an audience sub-form.
+ *
+ * @param $agent_data
+ *   The current agent data.
+ * @param $rule_parent_identifier
+ *   The identifier used for this rule, either the machine_name or "add" for
+ *   new rules.
+ * @param $rule
+ *   (Optional) The rule for an existing audience rule.  This contains:
+ *   - label: The name of the audience
+ *   - weight: the weight of the audience as compared ot all audiences for the
+ *     option set.
+ *   - targeting_features: An array of feature strings (context/value)
+ *   - targeting_rules: An array of rules including context and plugin
+ *   - targeting_strategy: The strategy to use when combining contexts
+ * @return array
+ *   The form element
+ */
+function _acquia_lift_personalize_campaign_wizard_targeting_audience(&$form, &$form_state, $agent_data, $rule_parent_identifier, $rule = NULL) {
+// Build up a list of available context values for targeting.
+  module_load_include('inc', 'personalize', 'personalize.admin');
+  $targeting_values = personalize_get_targeting_values_for_agent($agent_data);
+  $form = array(
+    '#type' => 'container',
+  );
+  $form['name'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Audience name'),
+    '#default_value' => $rule['label'],
+    '#required' => !empty($rule),
+  );
+  $form['rule_id'] = array(
+    '#type' => 'value',
+    '#value' => $rule_parent_identifier,
+  );
+
+  // This is the portion of the form that will be replace when the "add new" or "remove
+  // context" links are clicked.
+  $main_wrapper_id = 'acquia-lift-targeting';
```

Comment:
This needs to get the audience name appended to it or something - currently if you have two or more audiences you end up with multiple divs with this id and then the "Add context" button on the second audience results in a context being added to the first audience.